src/main/java/org/jenkinsci/plugins/durabletask/WindowsBatchScript.java

```diff
@@ -54,20 +55,40 @@ public String getScript() {
         }
         BatchController c = new BatchController(ws);
 
-        c.getBatchFile1(ws).write(String.format("cmd /c \"\"%s\"\" > \"%s\" 2>&1\r\necho %%ERRORLEVEL%% > \"%s\"\r\n",
-                c.getBatchFile2(ws).getRemote().replace("%", "%%"),
-                c.getLogFile(ws).getRemote().replace("%", "%%"),
-                c.getResultFile(ws).getRemote().replace("%", "%%")
-        ), "UTF-8");
+        /*
+          This wrapper script basically only calls the actual script that is supposed to be executed. It redirects all
+          the output of the target script to the log file and stores the exit code in a result file.
+          The script uses a additional temporary to store the exit code that is moved after writing to is is done.
+          Writing the output file directly may cause problems because the check for the exit code may locate the result
+          file after it was created but before the code was written in and issue a error because of this.
+         */
+        final String nl = "\r\n";
+        c.getBatchFile1(ws).write(
+                "CHCP 65001 > NUL" + nl +
```

Comment:
Correct. It shouldn't change much. But I wanted to be sure that the encoding of the command line context and the written files match.